python/foglamp/plugins/south/sinusoid/sinusoid.py

```diff
@@ -49,6 +48,111 @@
 _LOGGER = logger.setup(__name__, level=logging.INFO)
 index = -1
 _task = None
+loop = None
+c_callback = None
+c_ingest_ref = None
+
+sine = [
+    0.0,
+    0.104528463,
+    0.207911691,
+    0.309016994,
+    0.406736643,
+    0.5,
+    0.587785252,
+    0.669130606,
+    0.743144825,
+    0.809016994,
+    0.866025404,
+    0.913545458,
+    0.951056516,
+    0.978147601,
+    0.994521895,
+    1.0,
+    0.994521895,
+    0.978147601,
+    0.951056516,
+    0.913545458,
+    0.866025404,
+    0.809016994,
+    0.743144825,
+    0.669130606,
+    0.587785252,
+    0.5,
+    0.406736643,
+    0.309016994,
+    0.207911691,
+    0.104528463,
+    1.22515E-16,
+    -0.104528463,
+    -0.207911691,
+    -0.309016994,
+    -0.406736643,
+    -0.5,
+    -0.587785252,
+    -0.669130606,
+    -0.743144825,
+    -0.809016994,
+    -0.866025404,
+    -0.913545458,
+    -0.951056516,
+    -0.978147601,
+    -0.994521895,
+    -1.0,
+    -0.994521895,
+    -0.978147601,
+    -0.951056516,
+    -0.913545458,
+    -0.866025404,
+    -0.809016994,
+    -0.743144825,
+    -0.669130606,
+    -0.587785252,
+    -0.5,
+    -0.406736643,
+    -0.309016994,
+    -0.207911691,
+    -0.104528463
+]
+
+
+def generate_data():
+    global index
+    while index >= -1:
+        # index exceeds, reset to default
+        if index >= 59:
+            index = -1
+        index += 1
+        yield sine[index]
+
+async def save_data(handle):
+    try:
+        while True:
+            time_stamp = utils.local_timestamp()
+            data = {
+                'asset': handle['assetName']['value'],
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': {
+                    "sinusoid": next(generate_data())
+                }
+            }
+
+            _LOGGER.info("sinusoid: returning reading via ingest.ingest_callback, %s", handle['assetName']['value'])
+            ingest.ingest_callback(c_callback, c_ingest_ref, data)
+            _LOGGER.info("sinusoid: returned reading via ingest.ingest_callback")
+
+            try:
+                await asyncio.sleep(1 / (float(handle['dataPointsPerSec']['value'])))
+            except ZeroDivisionError:
+                _LOGGER.warning('Data points per second must be greater than 0, defaulting to 1')
+                handle['dataPointsPerSec']['value'] = '1'
```

Comment:
South C server add `readingsPerSec` to AdvanceConfig; what should be the valid config item in this case?